pyuvdata/uvdata/mir.py

```diff
@@ -0,0 +1,194 @@
+        sma_alt = 4080.0
+
+        # Use the mir_parser to read in metadata, which can be used to select data.
+        mir_data = mir_parser.MirParser(filepath)
+
+        # Select out data that we want to work with.
+        if isource is None:
+            isource = mir_data.in_read["isource"][0]
+        if irec is None:
+            irec = mir_data.bl_read["irec"][0]
+        if isb is None:
+            isb = mir_data.bl_read["isb"][0]
+        if corrchunk is None:
+            corrchunk = mir_data.sp_read["corrchunk"][0]
+
+        mir_data.use_in = mir_data.in_read["isource"] == isource
+        mir_data.use_bl = np.logical_and(
+            np.logical_and(
+                mir_data.bl_read["isb"] == isb, mir_data.bl_read["ipol"] == 0
+            ),
+            mir_data.bl_read["irec"] == irec,
+        )
+        mir_data.use_sp = mir_data.sp_read["corrchunk"] == corrchunk
+
+        # Load up the visibilities into the MirParser object. This will also update the
+        # filters, and will make sure we're looking at the right metadata.
+        mir_data._update_filter()
+        if len(mir_data.in_data) == 0:
+            raise IndexError("No valid records matching those selections!")
+
+        mir_data.load_data(load_vis=True, load_raw=True)
+
+        # Create a simple array/list for broadcasting values stored on a
+        # per-intergration basis in MIR into the (tasty) per-blt records in UVDATA.
+        bl_in_maparr = [mir_data.inhid_dict[idx] for idx in mir_data.bl_data["inhid"]]
+
+        # Derive Nants_data this value from baselines.
+        self.Nants_data = len(
+            np.unique(
+                np.concatenate((mir_data.bl_data["iant1"], mir_data.bl_data["iant2"]))
+            )
+        )
+
+        self.Nants_telescope = 8
+        self.Nbls = int(self.Nants_data * (self.Nants_data - 1) / 2)
+        self.Nblts = len(mir_data.bl_data)
+        self.Nfreqs = int(mir_data.sp_data["nch"][0])
+        self.Npols = 1  # todo: We will need to go back and expand this.
+        self.Nspws = 1  # todo: We will need to go back and expand this.
+        self.Ntimes = len(mir_data.in_data)
+        self.ant_1_array = mir_data.bl_data["iant1"] - 1
+        self.ant_2_array = mir_data.bl_data["iant2"] - 1
+        self.antenna_names = [
+            "Ant 1",
+            "Ant 2",
+            "Ant 3",
+            "Ant 4",
+            "Ant 5",
+            "Ant 6",
+            "Ant 7",
+            "Ant 8",
+        ]
+        self.antenna_numbers = np.arange(8)
+
+        # Prepare the XYZ coordinates of the antenna positions.
+        antXYZ = np.zeros([self.Nants_telescope, 3])
+        for idx in range(self.Nants_telescope):
+            if (idx + 1) in mir_data.antpos_data["antenna"]:
+                antXYZ[idx] = mir_data.antpos_data["xyz_pos"][
+                    mir_data.antpos_data["antenna"] == (idx + 1)
+                ]
+
+        self.antenna_positions = uvutils.ECEF_from_rotECEF(antXYZ, sma_lon)
```

Comment:
Karto will look into this once we finish all the other comments.

Reply:
So I checked this and it seems like we have coordinate values that are appropropriate, I think because we are going from _relative_ rotECEF to _relative_ ECEF (rather than absolute rotECEF to ECEF -- both are just z-axis rotations). On closer inspection, the antennas are referenced to the correct position (one antenna occupies that position, which makes it easy to check), so I think we've implemented this correctly.

Reply:
Yep, that's what we do in Miriad and UVFits.